### PR TITLE
Bug fixes for reports page

### PR DESCRIPTION
**Pull request by JayshreeFlyers**
- Added Loader for Plant and user table.
- Added pagination for users page,
- Added bulk upload button in mastery screen. 
- Changed the routes to mastery plant screen when clicked on the mastery on side nav. 
- Bug fixes for reports and mastery page. 

**_Screenshots:_** 
image
image

### CODE REVIEW

File: src/components/reusable/breadcrumbs/index.tsx [hunk 1]
```diff
@@ -54,6 +56,16 @@ function Breadcrumbs({ className, crumbs }: BreadcrumbsType) {
           />
         </div>
       ) : null}
+      {location.pathname === SITEMAP.mastery.manualEntry ? (
+        <div>
```
Comment (chandran-kannan) [inline src/components/reusable/breadcrumbs/index.tsx]: why this div added here?

File: src/pages/users/UsersList.tsx [hunk 2]
```diff
@@ -46,9 +49,11 @@ function UsersList() {
 
   // fetching users data by role
   const fetchUserDataByRole = async (page: number, limit: number) => {
+    setIsLoading(true);
     if (loggedUser) {
       const res = await USER_SERVICES.getAllUsers(page, limit);
```
Comment (chandran-kannan) [inline src/pages/users/UsersList.tsx:54]: move this inside if statement `setIsLoading(true);`

File: src/pages/users/UsersList.tsx [hunk 3]
```diff
@@ -46,9 +49,11 @@ function UsersList() {
...
       setUserData(res?.message);
+      setIsLoading(false);
     }
```
Comment (chandran-kannan) [inline src/pages/users/UsersList.tsx:59]: add else condition and show toast or console.error('user is not logged in')

File: src/pages/users/UsersList.tsx [hunk 4]
```diff
@@ -271,18 +276,10 @@ function UsersList() {
         className="w-full mx-auto"
         dataSource={userData}
         columns={columns}
-        // pagination={{
```
Comment (chandran-kannan) [inline src/pages/users/UsersList.tsx:274]: why paginations are removed

**Review by chandran-kannan — CHANGES_REQUESTED** [review]

File: src/pages/users/UsersList.tsx (hunk 4 above)
Comment (JayshreeFlyers) [reply src/pages/users/UsersList.tsx:274]: As Ant design table has default pagination so i removed the commented lines alone.

**Review by chandran-kannan — APPROVED** [review]

**Pull request merged**